src/main/kotlin/io/snyk/plugin/cli/Platform.kt

```diff
@@ -16,9 +17,8 @@ class Platform(val snykWrapperFileName: String) {
 
         @Throws(PlatformDetectionException::class)
         fun detect(systemProperties: Properties): Platform {
-            val architectureName = (systemProperties["os.name"] as String).toLowerCase(Locale.ENGLISH)
 
-            return when(architectureName) {
+            return when (val architectureName = (systemProperties["os.name"] as String).toLowerCase(Locale.ENGLISH)) {
```

Comment:
I usually asj to inline things, but here the original way to define val seems to be more readable... You've got that Kotlin minimisation desees :)

Reply:
Haha, I also wanted to get rid of the warning ;)